From af32ceae81be97ce99ab3146d1bf6b6a20f79e6d Mon Sep 17 00:00:00 2001
From: Florian Westphal <fwestpha@redhat.com>
Date: Tue, 2 Sep 2014 10:14:37 -0400
Subject: [net] icmp: Call skb_checksum_simple_validate

Message-id: <1409652920-431-15-git-send-email-fwestpha@redhat.com>
Patchwork-id: 90140
O-Subject: [PATCH 14/57] [RHEL7.1 net] icmp: Call skb_checksum_simple_validate
Bugzilla: 1131999
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Vlad Yasevich <vyasevic@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>

BZ: 1131999

commit 29a96e1f36dbd6fc7911a6d517625c656ba4809f
Author: Tom Herbert <therbert@google.com>
Date:   Wed May 7 16:52:21 2014 -0700

    icmp: Call skb_checksum_simple_validate

    Use skb_checksum_simple_validate to verify checksum.

    Signed-off-by: Tom Herbert <therbert@google.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/ipv4/icmp.c b/net/ipv4/icmp.c
index f6e4008..f9ce257 100644
--- a/net/ipv4/icmp.c
+++ b/net/ipv4/icmp.c
@@ -879,16 +879,8 @@ int icmp_rcv(struct sk_buff *skb)
 
  ICMP_INC_STATS_BH(net, ICMP_MIB_INMSGS);
 
- switch (skb->ip_summed) {
- case CHECKSUM_COMPLETE:
-  if (!csum_fold(skb->csum))
-   break;
-  /* fall through */
- case CHECKSUM_NONE:
-  skb->csum = 0;
-  if (__skb_checksum_complete(skb))
-   goto csum_error;
- }
+ if (skb_checksum_simple_validate(skb))
+  goto csum_error;
 
  if (!pskb_pull(skb, sizeof(*icmph)))
   goto error;
-- 
1.7.1